provider/core/src/error.rs

```diff
@@ -72,6 +72,9 @@ pub enum DataErrorKind {
     /// means that a required Cargo feature was not enabled
     #[displaydoc("Unavailable buffer format: {0:?} (does icu_provider need to be compiled with an additional Cargo feature?)")]
     UnavailableBufferFormat(BufferFormat),
+
+    /// The values for two [`DataKey`]s are not consistent.
```

Comment:
nit: please expand on this, on what "not consistent" can mean here

Reply:
I want to leave this kind of vague. Do you have a suggestion for wording?

Reply:
"The values for two datamarkers that deal with related data are not consistent"

perhaps have an example